File: leetcode/problems/problem-specific/557.reverse-words-in-a-string-iii/557.reverse-words-in-a-string-iii.go
package main

import "strings"

func reverseWords(s string) string {
	if len(s) <= 1 {
		return s
	}

	res, sSplit := "", strings.Split(s, " ")
	for _, v1 := range sSplit {
		for i := len(v1) - 1; i >= 0; i-- {
			res += string(v1[i])
		}

		res += " "
	}

	return res[:len(res)-1]
}

File: leetcode/problems/problem-specific/557.reverse-words-in-a-string-iii/557.reverse-words-in-a-string-iii.py
#
# @lc app=leetcode id=557 lang=python3
#
# [557] Reverse Words in a String III
#

# @lc code=start

# * Two Pointers Solution | O(n) Time | O(1) Space
# * n -> The length of s string


class Solution:
    def reverseWords(self, s: str) -> str:
        return " ".join([word[::-1] for word in s.split()])


# @lc code=end
